Lib/test/test_inspect.py

```diff
@@ -2321,6 +2321,138 @@ async def func(a=None):
                          {'a': None, 'gencoro': gencoro, 'b': 'spam'})
 
 
+class TestGetAsyncGenState(unittest.TestCase):
+
+    def setUp(self):
+        async def number_asyncgen():
+            for number in range(5):
+                yield number
+        self.asyncgen = number_asyncgen()
+
+    def tearDown(self):
+        try:
+            self.asyncgen.aclose().send(None)
+        except StopIteration:
+            pass
+
+    def _asyncgenstate(self):
+        return inspect.getasyncgenstate(self.asyncgen)
+
+    def test_created(self):
+        self.assertEqual(self._asyncgenstate(), inspect.AGEN_CREATED)
+
+    def test_suspended(self):
+        try:
+            next(anext(self.asyncgen))
+        except StopIteration as exc:
+            self.assertEqual(self._asyncgenstate(), inspect.AGEN_SUSPENDED)
+            self.assertEqual(exc.args, (0,))
```

Comment:
Add else clause with `self.fail` otherwise this test can be broken and will still pass.

Reply:
Good catch! After inheriting from `IsolatedAsyncioTestCase` adding the else-clause is not needed anymore.